server/src/services/search.service.ts

```diff
@@ -64,12 +67,20 @@ export class SearchService extends BaseService {
   }
 
   async searchRandom(auth: AuthDto, dto: RandomSearchDto): Promise<AssetResponseDto[]> {
+    if (dto.visibility === AssetVisibility.LOCKED && !auth.session?.hasElevatedPermission) {
```

Comment:
Maybe consider putting this in this access until with something like requireElevatedAccess